### PR TITLE
Axisfault is thrown when xml payload is sent with application/json co…

### PR DESCRIPTION
**Pull request by ravindraranwala**
…ntent type.

### CODE REVIEW

File: modules/commons/src/main/java/org/apache/synapse/commons/json/JsonStreamBuilder.java [hunk 1]
```diff
@@ -43,12 +43,15 @@ public OMElement processDocument(InputStream inputStream, String s,
...
-            }
+			if (element != null) {
+				if (logger.isDebugEnabled()) {
+					logger.debug("#processDocument. Built JSON payload from JSON stream. MessageID: " +
+					             messageContext.getMessageID());
+				}
+				return element;
+			} else {
+				throw new AxisFault("No JSON payload provided.");
+			}
```
Comment (Buddhima) [inline modules/commons/src/main/java/org/apache/synapse/commons/json/JsonStreamBuilder.java]: Please fix indentation issue here

File: modules/commons/src/main/java/org/apache/synapse/commons/json/JsonStreamBuilder.java [hunk 2]
```diff
@@ -43,12 +43,15 @@ public OMElement processDocument(InputStream inputStream, String s,
...
         if (inputStream != null) {
             OMElement element = JsonUtil.newJsonPayload(messageContext, inputStream, false, false);
-            if (element != null) {
-                if (logger.isDebugEnabled()) {
-                    logger.debug("#processDocument. Built JSON payload from JSON stream. MessageID: " + messageContext.getMessageID());
-                }
-                return element;
-            }
+			if (element != null) {
+				if (logger.isDebugEnabled()) {
```
Comment (nuwand) [inline modules/commons/src/main/java/org/apache/synapse/commons/json/JsonStreamBuilder.java]: could have made it a single statement by using &. 

if(element != null && logger.isDebugEnabled())

Comment (ravindraranwala) [inline modules/commons/src/main/java/org/apache/synapse/commons/json/JsonStreamBuilder.java]: I have a doubt here. Coz we need to return element if it is not null despite of enabling/disabling debug logs.

**Comment by ravindraranwala** [comment]
This pr is already merged to master branch by https://github.com/wso2/wso2-synapse/pull/284, hence closing the pr here.

**Pull request closed without merging**